File: test_inject_lib.py
"""Library for testing Python type hint injection.
These functions have NO type annotations — trickle will add them.
"""


def calculate_tax(amount, rate):
    tax = amount * (rate / 100)
    return {
        "amount": amount,
        "rate": rate,
        "tax": tax,
        "total": amount + tax,
    }


def format_user(user, locale="en-US"):
    return {
        "display": f"{user['first_name']} {user['last_name']}",
        "email": user["email"].lower(),
        "locale": locale,
    }


def filter_items(items, min_value):
    matching = [x for x in items if x > min_value]
    return {
        "results": matching,
        "count": len(matching),
        "total": len(items),
    }

File: tests/test-inject-lib.js
/**
 * Library for testing JSDoc injection.
 * These functions have NO type annotations — trickle will add JSDoc.
 */

function calculateTax(amount, rate) {
  const tax = amount * (rate / 100);
  return {
    amount,
    rate,
    tax,
    total: amount + tax,
  };
}

function formatUser(user, locale) {
  return {
    display: `${user.firstName} ${user.lastName}`,
    email: user.email.toLowerCase(),
    locale: locale || 'en-US',
  };
}

function filterItems(items, minValue) {
  const matching = items.filter(x => x > minValue);
  return {
    results: matching,
    count: matching.length,
    total: items.length,
  };
}

module.exports = { calculateTax, formatUser, filterItems };
